e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -1389,10 +1419,7 @@ def _get_year_str(self, year: int) -> str:
         str
             The year as a string (e.g., "2001", "0001").
         """
-        if year >= 0 and year < 1000:
-            return f"{year:04}"
-
-        return str(year)
+        return str(year).zfill(4)
```

Comment:
Fixed a bug where `end_yr_str` would be in the improper format for Xarray subsetting (e.g., `2-01-01`). `zfill(4)` pads the string to ensure it has 4 digits for the year (e.g., `0002-01-01`).